dbt_semantic_interfaces/implementations/metric.py

```diff
@@ -168,6 +173,18 @@ class PydanticCumulativeTypeParams(HashableBaseModel):
     period_agg: PeriodAggregation = PeriodAggregation.FIRST
 
 
+class PydanticMetricAsSourceNodeParams(HashableBaseModel):
```

Comment:
Alright let's talk naming! I feel like `MetricAsSourceNode` is confusing since we don't really use the terms "source" or "node" in this context. We do in MetricFlow, but there, a "source node" refers to the semantic model. The change here at a high level is that we're applying the aggregation at the metric level, instead of the measure level. So I'm thinking the name should have something to do with metric aggregation. I like `PydanticMetricAggregationParams` or something similar to that. (Same for the protocol)

Reply:
Noticing that this proposed name is weird when you see the class contains `PydanticMeasureAggregationParameters`... but I feel like that one was a bad name? Would love to change it to be percentile-specific since that's all it contains but I don't think it's worth dealing with the downstream implications of a breaking change here. If you hate those two names next to each other we could potentially recreate that class with a new name and use that here, but I'll leave it up to you on that front.